src/Native/Unix/System.Native/pal_time.c

```diff
@@ -48,75 +48,43 @@ int32_t SystemNative_UTimensat(const char* path, TimeSpec* times)
     return result;
 }
 
-int32_t SystemNative_GetTimestampResolution(uint64_t* resolution)
+uint64_t SystemNative_GetTimestampResolution()
 {
-    assert(resolution);
-
 #if HAVE_MACH_ABSOLUTE_TIME
     mach_timebase_info_data_t mtid;
-    if (mach_timebase_info(&mtid) == KERN_SUCCESS)
-    {
-        *resolution = SecondsToNanoSeconds * ((uint64_t)(mtid.denom) / (uint64_t)(mtid.numer));
-        return 1;
-    }
-    else
-    {
-        *resolution = 0;
-        return 0;
-    }
 
-#elif HAVE_CLOCK_MONOTONIC
+    kern_return_t result = mach_timebase_info(&mtid);
+    assert(result == KERN_SUCCESS);
```

Comment:
Why is it ok to assume this will always succeed?  Do we know that if it fails it won't write garbage or incorrect values into mtid?

Reply:
I actually can't find any docs on the API, I'll change it back to explicitly returning zero on failure.

Reply:
Fixed.